src/app/features/search/services/search/search.service.ts

```diff
@@ -0,0 +1,32 @@
+import { HttpClient } from '@angular/common/http';
+import { inject, Injectable, signal } from '@angular/core';
+import { CityInfo } from '@features/search/interfaces/city-info';
+import { catchError } from 'rxjs/operators';
+import { of } from 'rxjs';
+import { SearchRouteParams } from '../../interfaces/search-route-params';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SearchService {
+  private _cities = signal<CityInfo[]>([]);
+
+  public cities = this._cities.asReadonly();
+
+  private httpClient = inject(HttpClient);
+
+  search({ fromLatitude, fromLongitude, toLatitude, toLongitude, time }: SearchRouteParams) {
+    const params = {
+      fromLatitude,
+      fromLongitude,
+      toLatitude,
+      toLongitude,
+      time,
+    };
+    return this.httpClient.get('/api/search', { params }).pipe(catchError((error) => of(error)));
```

Comment:
`catchError` is useless here, we should return some real value in it, please check https://dev.to/georgeknap/the-catch-in-the-rxjs-catcherror-51mp

Reply:
https://thabo-ambrose.medium.com/differentiating-between-catcherror-throwerror-new-error-and-throw-new-error-94555ec62a53

Reply:
Thanks for materials